R/importData.R

```diff
@@ -12,46 +12,56 @@
 #'
 #' @author Hagen Tockhorn
 #'
-#' @importFrom stringr str_split
+#' @importFrom stringr str_split str_detect
 #' @importFrom terra rast subset aggregate ext res
 #' @importFrom ncdf4 nc_open
+#' @importFrom madrat getConfig
+#' @importFrom utils tail
 #'
 #' @note
 #' folder structure in inputdata/sources/ISIMIPbuildings is expected to be:
 #'    country masks : var/
 #'    population :    var/scenario
 #'    other :         var/scenario/model
 #'
-#' @note currently, this function only reads data from ISIMIP3b
+#' @note currently, this function only reads data from ISIMIP3a and ISIMIP3b
 
 
-readISIMIPbuildings <- function(subtype) {
+importData <- function(subtype) {
+  # SOURCE DIRECTORY -----------------------------------------------------------
+
+  # check if source folder exists
+  sourceDir <- getConfig("sourcefolder")
+  if (!dir.exists(sourceDir)) {
+    stop("No madrat source directory found. Please define with `madrat::setConfig(\"sourcefolder\") = ...`")
+  }
+
+  sourceDir <- file.path(sourceDir, "ISIMIPbuildings")
 
-  # PARAMETERS------------------------------------------------------------------
+
+
+  # PARAMETERS -----------------------------------------------------------------
 
   baitVars <- c("tas", "sfcwind", "rsds", "huss")
 
   firstHistYear <- 1960
 
 
-  # FUNCTIONS-------------------------------------------------------------------
+
+  # FUNCTIONS ------------------------------------------------------------------
 
   splitSubtype <- function(subtype) {
     vars <- list()
 
     # nolint start
```

Comment:
Please remove nolint. I guess you don't need it anymore after removing the linebreaks before `else`.